terraform/main.tf

```diff
@@ -0,0 +1,23 @@
+module "pi_hole" {
+  source                     = "./pi-hole"
+  aws_credentials_profile    = var.aws_credentials_profile
+  aws_region                 = var.aws_region
+  ingress_access_ip_address  = var.ingress_access_ip_address
+  pi_hole_key_pair_name      = var.pi_hole_key_pair_name
+  pi_hole_web_admin_password = var.pi_hole_web_admin_password
+  enable_r53_zone_code       = var.enable_r53_zone_code
+  r53_zone                   = var.r53_zone
+  additional_tags            = var.additional_tags
+  open_vpn_source_ip         = module.open_vpn.public_ip
+}
+
+module "open_vpn" {
```

Comment:
Would like to see both pi-hole and open-vpn optional

Reply:
i looked at doing this... you cant have "optional" modules though. You cant use count : 0?! args etc in a module declaration. only option i looked at which wasnt all that neat. Was pass a bool into each module ..... and then add a count check in every resource inside the module. but that causes some issues with having to [0] array append resource refs. so it became really un-neat and ugly

Reply:
open to suggestions

Reply:
I thought you could use count in version 0.13 modules now?
https://www.hashicorp.com/blog/announcing-hashicorp-terraform-0-13

Reply:
will try this again .. I am definitely on the new 0.13 of tf.... and i tried a module based count call. and it didnt validate. count argument unexpected here ... error was thrown each way i tried it.. Hmmmm. I will re implement and have a look at the docs and see since i really did want this optional setup too

Reply:
might be introduced in the new 0.13.4 instead of 0.13.3

Reply:
Looks like it was released yesterday
https://github.com/hashicorp/terraform/blob/v0.13/CHANGELOG.md

Reply:
i got module optionals working. now i just have the issue that if you choose not to have either ec2 instance the 

```
output "pi_hole_r53_entries" {
  description = "Pi Hole DNS Entries"
  value       = module.open_vpn[0].r53_entry
}
```

references like this that use the module ouputs all blow up 

i have tries a `join("", module.open_vpn[0].r53_entry)` but that also doesnt work as it still tries to evaluate [0]